src/main/java/org/frc/team4048/rosnetworktables/NtRosProxy.java

```diff
@@ -2,105 +2,74 @@
-import java.net.InetAddress;
-import java.net.URI;
 import java.net.URISyntaxException;
-import java.net.UnknownHostException;
-import java.util.List;
-
-import static java.lang.Thread.sleep;
 
 public class NtRosProxy {
-     private static NtRosProxy instance;
-
-     public synchronized static NtRosProxy get() {
-          if(instance == null) instance = new NtRosProxy();
-          return instance;
-     }
-
-     private RosNode rosNode;
-     private final String rosMasterURI;
-     private final String rosHostname;
-     private final String networkTablesIP;
-     private Topics topics;
-     private NetworkTableInstance ntInstance;
-     private NetworkTable ntTable;
-     private boolean started = false;
-
-     private NtRosProxy() {
-          String rosMasterURI = System.getenv(Constants.ROS_MASTER_URI);
-          assert rosMasterURI != null : "Environment Variable not set " + Constants.ROS_MASTER_URI;
+    private static NtRosProxy instance;
 
-          String networkTablesIP = System.getenv(Constants.NT_IP);
-          assert networkTablesIP != null : "Environment Variable not set " + Constants.NT_IP;
+    public synchronized static NtRosProxy get() {
+        if (instance == null) instance = new NtRosProxy();
+        return instance;
+    }
 
-          String tempHostName = System.getenv(Constants.ROS_IP);
-          assert tempHostName != null : "Environment Variable not set " + Constants.ROS_IP;
+    private JRos2Client rosclient;
+    private final String networkTablesIP;
+    private Topics topics;
+    private NetworkTableInstance ntInstance;
+    private NetworkTable ntTable;
+    private boolean started = false;
 
-          this.rosMasterURI = rosMasterURI;
-          this.networkTablesIP = networkTablesIP;
-          this.rosHostname = tempHostName;
-     }
+    private NtRosProxy() {
+        String networkTablesIP = System.getenv(Constants.NT_IP);
+        if (networkTablesIP == null) {
+            throw new IllegalArgumentException("Environment Variable not set " + Constants.NT_IP);
+        }
+        this.networkTablesIP = networkTablesIP;
+    }
 
-     //TODO Add timeout
-     public void start() throws InterruptedException, URISyntaxException, IOException, ClassNotFoundException, InvocationTargetException, InstantiationException, IllegalAccessException {
-          if(started) return;
-          started = true;
-          initRosNode();
-//      Prints waiting every second while we wait for rosNode to init
-//          TODO add time out
-          long st = System.currentTimeMillis();
-          while (!rosNode.isInitialized()){
-               if(System.currentTimeMillis()-st >=1000){
-                    System.out.println("WAITING");
-                    st=System.currentTimeMillis();
-               }
-               sleep(100);
-          }
-          initNetworkTables();
-          initializeTopics();
-          topics.start();
-     }
+    //TODO Add timeout
```

Comment:
While you're in this code, is this comment legitimate? What would you be timing out?

Reply:
Done.